homeWork5/lesson1/Program.cs

```diff
@@ -0,0 +1,449 @@
+/// Переопределенный метод Equals() 
+/// </summary>
+/// <param name="obj">object</param>
+/// <returns>bool</returns>
+    public override bool Equals(object? obj)
+    {
+        if (obj is not RatinalNumbers rnumbers)
+        {
+           return false;
+        }
+        return Numerator == rnumbers.Numerator &&
+               Denumenator == rnumbers.Denumenator;
+    }
+
+/// <summary>
+/// Переопределенный метод GetHashCode()
+/// </summary>
+/// <returns>int</returns>
+    public override int GetHashCode()
+    {
+        int hashcode = _numerator.GetHashCode();
+        hashcode = 31 * hashcode + _denumerator.GetHashCode();
+        return hashcode;
+    }
+
+/// <summary>
+/// Переопределенный метод ToString()
+/// </summary>
+/// <returns>string</returns>
+    public override string ToString()
+    {
+        return $"{ _numerator }/{ _denumerator }";
+    }
+
+    /// <summary>
+    /// Переопределенный оператор приведения float
+    /// </summary>
+    /// <param name="rnumbers">RatinalNumbers</param>
+    public static explicit operator float(RatinalNumbers rnumbers)
+    {
+        return (float)rnumbers.Numerator / rnumbers.Denumenator;
+    }
+
+    /// <summary>
+    /// Переопределенный оператор приведения int
+    /// </summary>
+    /// <param name="rnumbers">RatinalNumbers</param>
+    public static explicit operator int(RatinalNumbers rnumbers)
+    {
+        return (int)rnumbers.Numerator / rnumbers.Denumenator;
+    }
+
+    /// <summary>
+    /// Переопределенный оператор ==
+    /// </summary>
+    /// <param name="rnumbers1">RatinalNumbers</param>
+    /// <param name="rnumbers2">RatinalNumbers</param>
+    /// <returns>bool</returns>
+    public static bool operator ==(RatinalNumbers rnumbers1, RatinalNumbers rnumbers2)
+    {
+        if (ReferenceEquals(rnumbers1, rnumbers2))
+        {
+            return true;
+        }
+            
+        if (ReferenceEquals(rnumbers1, null)) 
+        {
+            return false;
+        }
+            
+        if (ReferenceEquals(rnumbers2, null))
+        {
+            return false;
+        }
+        return rnumbers1.Equals(rnumbers2);
+    }
+
+    /// <summary>
+    /// Переопределенный оператор !=
+    /// </summary>
+    /// <param name="rnumbers1">RatinalNumbers</param>
+    /// <param name="rnumbers2">RatinalNumbers</param>
+    /// <returns>bool</returns>
+    public static bool operator !=(RatinalNumbers rnumbers1, RatinalNumbers rnumbers2) => !(rnumbers1 == rnumbers2);
+
+    /// <summary>
+    /// Переопределенный оператор <
+    /// </summary>
+    /// <param name="rnumbers1">RatinalNumbers</param>
+    /// <param name="rnumbers2">RatinalNumbers</param>
+    /// <returns>bool</returns>
+    public static bool operator <(RatinalNumbers rnumbers1, RatinalNumbers rnumbers2)
+    {
+        if (rnumbers1.Numerator / rnumbers1.Denumenator < rnumbers2.Numerator / rnumbers2.Denumenator)
+        {
+            return true;
+        }
+        return false;

```

Comment:
Индусский код!
```suggestion
        return rnumbers1.Numerator / rnumbers1.Denumenator < rnumbers2.Numerator / rnumbers2.Denumenator;
```
Почему для сравнения двух дробей не привести их в десятичную форму и не сравнить два вещественных числа? Зачем мудрить?